samples/csharp_dotnetcore/30.asp-mvc-bot/Asp-Mvc-Bot/BotActivityControllerBase.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Integration;
+using Microsoft.Bot.Configuration;
+using Microsoft.Bot.Schema;
+
+namespace Asp_Mvc_Bot
+{
+    /// <summary>
+    /// It is common practice to switch on the type of inbound Activity in the bot logic. This class implements that
```

Comment:
I think a comment better explaining how and why you would use this base class would be good here.